src/test/scala/cromwell/engine/db/slick/SlickDataAccessSpec.scala

```diff
@@ -0,0 +1,256 @@
+package cromwell.engine.db.slick
+
+import java.util.UUID
+
+import cromwell.binding._
+import cromwell.binding.command.Command
+import cromwell.binding.types.WdlStringType
+import cromwell.binding.values.WdlString
+import cromwell.engine._
+import cromwell.engine.backend.local.LocalBackend
+import cromwell.engine.db.DataAccess.WorkflowInfo
+import cromwell.engine.db.LocalCallBackendInfo
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{Millis, Seconds, Span}
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.concurrent.ExecutionContext
+
+class SlickDataAccessSpec extends FlatSpec with Matchers with ScalaFutures {
+  implicit val ec = ExecutionContext.global
+
+  implicit val defaultPatience = PatienceConfig(timeout = Span(5, Seconds), interval = Span(100, Millis))
+
+  // Tests against main database used for command line
+  "SlickDataAccess (main.hsqldb)" should behave like databaseWithConfig("main.hsqldb", testRequired = true)
+
+  // Tests using liquibase, but in memory
+  "SlickDataAccess (test.hsqldb)" should behave like databaseWithConfig("test.hsqldb", testRequired = true)
+
+  // If able to connect, then also run the tests on mysql, but it's not required
+  "SlickDataAccess (test.mysql)" should behave like databaseWithConfig("test.mysql", testRequired = false)
+
+  def databaseWithConfig(path: => String, testRequired: => Boolean): Unit = {
+
+    lazy val testDatabase = new TestSlickDatabase(DatabaseConfig.rootDatabaseConfig.getConfig(path))
+    lazy val dataAccess = testDatabase.slickDataAccess
+    lazy val canConnect = {
+      DatabaseConfig.rootDatabaseConfig.hasPath(path) && dataAccess.isValidConnection(1).futureValue
+    }
+
+    it should "setup via liquibase if necessary" in {
+      assume(canConnect || testRequired)
+      if (testDatabase.useLiquibase)
+        testDatabase.setupLiquibase()
+    }
+
+    val workflowIdForCreate = UUID.randomUUID()
+    val workflowInfoForCreate = new WorkflowInfo(workflowIdForCreate, "source", "{}")
+
+    it should "create a workflow for just reading" in {
+      assume(canConnect || testRequired)
+      dataAccess.createWorkflow(workflowInfoForCreate, Seq.empty, Seq.empty, new LocalBackend).futureValue
+    }
+
+    it should "retrieve the workflow for just reading" in {
```

Comment:
As written these tests appear to be interdependent.  Could the "create a workflow for just reading" be pulled into a setup method?